mllib/src/test/scala/org/apache/spark/ml/clustering/BisectingKMeansSuite.scala

```diff
@@ -51,6 +54,23 @@ class BisectingKMeansSuite
     assert(copiedModel.hasSummary)
   }
 
+  test("SPARK-16473: Verify Bisecting K-Means does not fail in edge case where" +
+    "one cluster is empty after split") {
+    val bkm = new BisectingKMeans().setK(k).setMinDivisibleClusterSize(4).setMaxIter(4)
+
+    assert(bkm.getK === k)
+    assert(bkm.getFeaturesCol === "features")
```

Comment:
There's no need to test featuresCol, predictionCol, and other things which aren't relevant to this unit test.  I'd simplify it.